Repository/NpmRepository.php

```diff
@@ -147,10 +148,17 @@ protected function createArrayRepositoryConfig(array $packageConfigs)
         $loader = new ArrayLoader();
 
         foreach ($packageConfigs as $version => $config) {
-            $config['version'] = $version;
-            $config = $this->assetType->getPackageConverter()->convert($config);
-            $config = $this->assetRepositoryManager->solveResolutions($config);
-            $packages[] = $loader->load($config);
+            try {
+                $config['version'] = $version;
+                $config = $this->assetType->getPackageConverter()->convert($config);
+                $config = $this->assetRepositoryManager->solveResolutions($config);
+                $packages[] = $loader->load($config);
+            } catch (\UnexpectedValueException $exception) {
+                // Most probably version constraint is broken.
+                // Skip this version and hope that another one will be OK
+                $this->io->write("<warning>Skipped {$config['name']} version {$version}: {$exception->getMessage()}</warning>", IOInterface::VERBOSE);
+                continue;
```

Comment:
- `continue` is not necessary
- remove comments because the output message is explicit

And it will be good for me.

Reply:
> remove comments because the output message is explicit

the message says much less than the comment. should probably improve the message to say, why the version was skipped.

Reply:
`$exception->getMessage()` will include the exact reason, e.g. "Could not parse version constraint >=1.1.0,,<2.0: Invalid version string "1.1.0,,<2.0"

Reply:
I'm for keeping the comment then, it does not hurt and provides more context.